logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/Message.kt

```diff
@@ -4,24 +4,46 @@ import com.wire.kalium.logic.data.conversation.ClientId
 import com.wire.kalium.logic.data.id.ConversationId
 import com.wire.kalium.logic.data.user.UserId
 
-data class Message(
-    val id: String,
-    val content: MessageContent,
-    val conversationId: ConversationId,
-    val date: String,
-    val senderUserId: UserId,
-    val senderClientId: ClientId,
-    val status: Status,
-    val editStatus : EditStatus,
-    val visibility: Visibility = Visibility.VISIBLE
+@Suppress("LongParameterList")
+sealed class Message(
+    open val id: String,
+    open val content: MessageContent,
+    open val conversationId: ConversationId,
+    open val date: String,
+    open val senderUserId: UserId,
+    open val status: Status,
+    open val visibility: Visibility
 ) {
+
+    data class Client(
+        override val id: String,
+        override val content: MessageContent.Client,
+        override val conversationId: ConversationId,
+        override val date: String,
+        override val senderUserId: UserId,
+        val senderClientId: ClientId,
+        override val status: Status,
+        val editStatus : EditStatus,
+        override val visibility: Visibility = Visibility.VISIBLE
```

Comment:
Maybe move the unique `Client` parameters to the end to make it clearer they are not shared with `Server`?
```suggestion
        override val visibility: Visibility = Visibility.VISIBLE,
        override val status: Status,
        val senderClientId: ClientId,
        val editStatus : EditStatus
```

Reply:
Good idea, I changed it 🚀